x-pack/plugins/apm/server/routes/fleet/create_cloud_apm_package_policy.ts

```diff
@@ -21,25 +21,28 @@ import {
   APMPluginStartDependencies,
 } from '../../types';
 import { getApmPackagePolicyDefinition } from './get_apm_package_policy_definition';
-import { Setup } from '../../lib/helpers/setup_request';
 import { mergePackagePolicyWithApm } from './merge_package_policy_with_apm';
 import { ELASTIC_CLOUD_APM_AGENT_POLICY_ID } from '../../../common/fleet';
+import { ApmIndicesConfig } from '../settings/apm_indices/get_apm_indices';
+import { APMInternalESClient } from '../../lib/helpers/create_es_client/create_internal_es_client';
 
 export async function createCloudApmPackgePolicy({
   cloudPluginSetup,
   fleetPluginStart,
   savedObjectsClient,
   esClient,
   logger,
-  setup,
+  indices,
+  internalESClient,
   request,
 }: {
   cloudPluginSetup: APMPluginSetupDependencies['cloud'];
   fleetPluginStart: NonNullable<APMPluginStartDependencies['fleet']>;
   savedObjectsClient: SavedObjectsClientContract;
   esClient: ElasticsearchClient;
   logger: Logger;
-  setup: Setup;
+  indices: ApmIndicesConfig;
+  internalESClient: APMInternalESClient;
```

Comment:
For the `apmEventClient` we can access `indices` via `apmEventClient.indices` ([example](https://github.com/elastic/kibana/blob/282b3d4f733009e672e3315aad7d9553611d2d74/x-pack/plugins/apm/server/routes/data_view/create_static_data_view.ts)). We already talked about it but for consistency I think we can do the same here. Another approach is to specify indices using strings that are then mapped to the correct index.